### PR TITLE
fix: Implement private only check on socket connect

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?
Implement private only check on socket connect

### CODE REVIEW

File: lib/realtime_web/channels/realtime_channel.ex [hunk 1]
```diff
@@ -676,4 +684,13 @@ defmodule RealtimeWeb.RealtimeChannel do
   defp maybe_assign_policies(_, _, _, _, socket) do
     {:ok, assign(socket, policies: nil)}
   end
+
+  defp only_private?(tenant_id, %{assigns: %{check_authorization?: check_authorization?}}) do
```
Comment (abc3) [inline lib/realtime_web/channels/realtime_channel.ex:688]:
```suggestion
  @spec only_private?(String.t(), map()) :: :ok | {:error, :private_only}
  defp only_private?(tenant_id, %{assigns: %{check_authorization?: check_authorization?}}) do
```

File: lib/realtime/api/tenant.ex [hunk 2]
```diff
@@ -74,7 +74,8 @@ defmodule Realtime.Api.Tenant do
       :max_channels_per_client,
       :max_joins_per_second,
       :suspend,
-      :notify_private_alpha
+      :notify_private_alpha,
+      :private_only
```
Comment (abc3) [inline lib/realtime/api/tenant.ex:78]: double checking if a migration is not needed for this

Comment (filipecabaco) [reply lib/realtime/api/tenant.ex:78]: migrations was in a different PR and has defaults to false

**Review by abc3 — APPROVED** [review]

**Pull request merged**